plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -187,10 +188,39 @@ func getTags(cert *x509.Certificate, location string) map[string]string {
 	return tags
 }
 
+// copied from plugins/inputs/file/file.go
+func (c *X509Cert) refreshFilePaths() error {
+	var allFiles []string
+
+	for _, source := range c.Sources {
+		if strings.HasPrefix(source, "/") {
+			g, err := globpath.Compile(source)
+			if err != nil {
+				return fmt.Errorf("could not compile glob %v: %v", source, err)
+			}
+			files := g.Match()
+			if len(files) <= 0 {
+				return fmt.Errorf("could not find file: %v", source)
+			}
+			allFiles = append(allFiles, files...)
+		} else {
+			allFiles = append(allFiles, source)
```

Comment:
Why do we skip glob compilation if the path isn't absolute?  It seems that we can just pass any string into globpath.Compile.

Reply:
Well that seems to be the original logic of this plugin, lines 60 to 63:
```func (c *X509Cert) locationToURL(location string) (*url.URL, error) {
	if strings.HasPrefix(location, "/") {
		location = "file://" + location
	}

Reply:
Ahh, strings not starting with `/` are URLs, I forgot we are handling both here.

Would you be able to move the matching into Init along with the URL parsing?  We could have separate fields on the struct for each type of path, file paths as a []GlobPath and URLs as a []*url.URL.  Then during gather we can call Match() on the globs and use the http URLs as they are.

Reply:
We could get rid of those `strings.HasPrefix` by forcing users to use `file://` url format in configuration, but that would break configuration's backward compatibility, I don't know if this is acceptable.
Also matching a leading `/` will probably not work for windows users right?